### DISCUSSION TITLE
v3에서 v4로 버전 업그레이드 중인데 메세지 가리기 함수는 뭔지 안보이네요..

### DISCUSSION THREAD

**Discussion by gangdaegu** [storycraft/node-kakao#809] [Q&A]
혹시 api 문서에서 어디를 보면 될까요?
제가 본 곳은 talk/openlink 쪽인데 혹시 어디를 봐야 할까요...??

가리기랑 삭제하기 함수를 알고 싶은데
어디를 봐야할지 모르겠습니다..

sendChat 함수를 따라가보니 deleteChat 이라는 함수도 발견하긴 했는데
이건 가리는게 아닌 삭제하는 함수인거 같아서요..

**Comment by tadashi74** [comment]
`if(data.text === '가리기테스트') channel.hideChat(data.chat);
`

**Reply by gangdaegu** [reply]
'TalkChannel' 형식에 'hideChat' 속성이 없습니다. 다고 떠서
hideChat을 사용할 수가 없더라구요..

**Reply by tadashi74** [reply]
> 'TalkChannel' 형식에 'hideChat' 속성이 없습니다. 다고 떠서 hideChat을 사용할 수가 없더라구요..

ㅋㅋㅋ 저랑 똑같은 문제를 겪으셨군요 https://github.com/storycraft/node-kakao/issues/530

`(chanenl as unknown as TalkOpenChannel).hideChat();`

로 하시거나 아니면 

```
CLIENT.channelList.open.on('chat', (data: TalkChatData, channel: TalkOpenChannel) => {
//오픈채팅방에서 오는 메시지에만 반응
if(data.text === '가리기테스트') channel.hideChat(data.chat);
});

```
로 하시면 될거에요 :)

**Reply by gangdaegu** [reply]
> > 'TalkChannel' 형식에 'hideChat' 속성이 없습니다. 다고 떠서 hideChat을 사용할 수가 없더라구요..
> 
> ㅋㅋㅋ 저랑 똑같은 문제를 겪으셨군요 #530
> 
> `(chanenl as unknown as TalkOpenChannel).hideChat();`
> 
> 로 하시거나 아니면
> 
> ```
> CLIENT.channelList.open.on('chat', (data: TalkChatData, channel: TalkOpenChannel) => {
> //오픈채팅방에서 오는 메시지에만 반응
> if(data.text === '가리기테스트') channel.hideChat(data.chat);
> });
> ```
> 
> 로 하시면 될거에요 :)

오오 덕분에 해결되었습니다! 감사합니다!
